### ISSUE TITLE
add a new test to disassemble+reassemble a full file and check the checksum of these files for each architectures

### ISSUE THREAD

**Issue by gogo2464** [radareorg/radare2#19481]
## Description

*Please describe what are you missing or wanting to be improved*

I was going to push the test for travis-ci/cd for a decompilation project for a 8051 firmware when I found a bug on recompilation: 

> Cannot assemble 'jbc 0x20.0, 0x8018' at line 36342

I also have 4 invalid instructions during the disassembly step.

I thing implementing tests over this way is a good deal to catch almost any bug on disassembling/reassembling. It should not even miss any single instruction.

I also would like to use r2 like that to disassemble/document/decompile old architecture. I currently can not really due to bug and people want to use another asm compiler...

*Provide images, ascii-art, test files and anything that may help us understand your request*

```
#!/bin/bash

radare2 -a 8051 -m 0x8000 -e "scr.color=0" -qf -c "pI" MP_8192E_FW_NIC.bin > out.asm
rasm2 -a 8051 -f out.asm > recompiled-firmware.bin

export old_bin=$(md5sum MP_8192E_FW_NIC.bin) ;
export new_bin=$(md5sum recompiled-firmware.bin) ;

if [ "$old_bin" = "$new_bin" ]
then
  echo "OK"
else
  echo "NOT OK"
  exit 1 ;
fi
```

**Comment by trufae** [comment]
yep. that should be possible and its planned, but there's no serious effort towards this beside fixing the assembler when there's a missing instruction or alignment issue or so

**Comment by Lazula** [comment]
i see no use for a checksum as opposed to a direct diff check: `[ ! $(diff file1 file2) ]`

no need to process the entire files if a mismatch occurs early on

**Comment by gogo2464** [comment]
@Lazula yes this could be a great update. This is actually another project than radare2. Could we talk about with discord instead please?

**Comment by trufae** [comment]
agree with @Lazula a checksum is more of a problem than a real benefit, for the developer and the user.
